soundcloud.py

```diff
@@ -0,0 +1,21 @@
+import requests
+
+from config import Config
+from serviceresult import ServiceResult
+
+class Soundcloud:
+    config: Config = Config()
+    Threshold = 500000
+    
+    def get_service_result(self, url):
+        # Follow URL to the end location in case of URL shorteners
+        session = requests.Session()  # so connections are recycled
+        resp = session.head(url, allow_redirects=True)
+        final_url = resp.url
+        
+        if 'soundcloud.com' not in final_url:
+            return ServiceResult(False, 0, 0, '')
+        
+        response = requests.get('https://api.soundcloud.com/resolve.json?url=' + url + '&client_id=' + self.config.SOUNDCLOUD_KEY)
```

Comment:
If our client ID is invalid what happens here?

Maybe test with an invalid ID.  Might need to wrap this with a try/catch and return a ServiceResult(False, 0, 0, '') if it fails.

Reply:
It will return the same 404 error no matter what kind of incorrect url is input. I have updated the code to work properly in this case. The commit is a bit weird, but i just added a couple lines around line 20 somewhere

Reply:
Now should work for all errors, both url error and client id error. Resolving